### PR TITLE
Rename section block to group

### LINKED ISSUE #14898: Renaming the new "Section" block to "Group"
Coming a bit late to the conversation around the new **Section** block (#13964). I don't like to resurface things that have already reached a consensus, however, I think the name for this block needs to change for a few reasons:

* I agree with the concerns raised before about the confusion with the semantic nature of the HTML5 element. Even without that potential confusion, “creating a section” carries a bit more weight than "creating a group" for stylistic purposes, and I think this block conflates the two.
* Once actual page sections start to appear to power other areas of the site/theme (headers, navigation, featured stories, sidebars, footers, and so on) it might lead to yet more semantic overlap.
*  People might not know when it is appropriate to add a section or not, and what implications it might have for their content structure. It should be fine to add a group within a _Column_ and so on without hesitating as to whether a section is called for.

I'd propose calling this block **Group** instead, which seems more straightforward and less loaded as a term. (In the future, there'd be space for supporting specific HTML tags for the containing element, and converting a Group to a proper Section seems like a meaningful path.)

---

Finally, I'd strongly suggest incorporating a way to select multiple blocks and grouping them by using the block transformation tool — select a few blocks → open transformations → choose the "Group" block.

I believe it can be a more natural pattern for people to first create their content, then group it, rather than start with an empty container first.

### PR DESCRIPTION
**Pull request by talldan**
## Description
Closes #14898

Renames the Section block to Group.

## How has this been tested?
- Manual testing
- Updated E2E tests

See also https://github.com/WordPress/gutenberg/pull/13964#issue-254541645 for further details on testing (but replace the word section with group in your mind).

## Screenshots 
Screen Shot 2019-04-11 at 4 23 54 pm

## Types of changes
Breaking change (but one that should be ok if it's merged before the next release)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by talldan** [comment]
@kjellr - I think you might have been working on some patches for themes so that they work with what was the section block. I suppose this change affects that and the classes will have to be renamed?

File: packages/block-library/src/editor.scss
Comment (gziolo) [inline packages/block-library/src/editor.scss]: Can it be moved to be sorted alphabetically?

File: packages/block-library/src/index.js
Comment (gziolo) [inline packages/block-library/src/index.js]: Can it be moved to be sorted alphabetically?

File: packages/e2e-tests/fixtures/block-transforms.js
Comment (gziolo) [inline packages/e2e-tests/fixtures/block-transforms.js]: Can it be moved to be sorted alphabetically?

**Review by gziolo — COMMENTED** [review]
It looks like a lot of work to rename a single block 😅

File: packages/block-library/src/editor.scss
Comment (talldan) [reply packages/block-library/src/editor.scss]: 👍 done

File: packages/block-library/src/index.js
Comment (talldan) [reply packages/block-library/src/index.js]: 👍 done

File: packages/e2e-tests/fixtures/block-transforms.js
Comment (talldan) [reply packages/e2e-tests/fixtures/block-transforms.js]: 👍 done

**Comment by kjellr** [comment]
> @kjellr - I think you might have been working on some patches for themes so that they work with what was the section block. I suppose this change affects that and the classes will have to be renamed?

Yep, that's right. None of the patches have been merged in yet, so this is a great time to make the change. 🙂

**Review by gziolo — APPROVED** [review]
The block itself can be inserted into post properly 👍

**Comment by chrisvanpatten** [comment]
I love this. When I had taken my stab at the container block, this was the approach I was trying to take, along with the "Group" and "Ungroup" functionality. I think this makes so much sense, makes it more intuitive for users, and really opens the door for a better user experience!

**Comment by gziolo** [comment]
What it's confusing at the moment is that when you insert Group block, what you actually see is Paragraph block:

Screen Shot 2019-04-11 at 17 17 37

Screen Shot 2019-04-11 at 17 18 21

**Comment by aduth** [comment]
> What it's confusing at the moment is that when you insert Group block, what you actually see is Paragraph block:

To be clear, should we consider this a separate task from the rename itself happening here? 

I expect the work from #14241 should help to improve the experience of the default block appender.

**Comment by getdave** [comment]
> I expect the work from #14241 should help to improve the experience of the default block appender.

Indeed my work on `core/section` was exactly what prompted me to start work on the placeholder/appender. I very much recommend we utilise this in both `core/group` and `core/column` once I can get that branch merged.

**Comment by getdave** [comment]
> I love this. When I had taken my stab at the container block, this was the approach I was trying to take, along with the "Group" and "Ungroup" functionality. I think this makes so much sense, makes it more intuitive for users, and really opens the door for a better user experience!

@chrisvanpatten Here's my first pass at "Group" https://github.com/WordPress/gutenberg/pull/14908/

**Comment by gziolo** [comment]
I was sharing my experience from testing. I didn’t mean it should block this PR 👍

**Comment by aduth** [comment]
> I was sharing my experience from testing. I didn’t mean it should block this PR 👍

Do you know if there's a related issue for tracking these improvements? Is #9628 relevant?

**Comment by gziolo** [comment]
> Do you know if there's a related issue for tracking these improvements? Is #9628 relevant?

I'm not 100 sure but I commented regardless :) I also cross-posted to #14935 to keep discussion going.

**Comment by getdave** [comment]
Great to see this rename get done so quickly @talldan 👍

**Pull request merged**
